neuroptica/optimizers.py

```diff
@@ -200,6 +208,21 @@ def fit(self, data: np.ndarray, labels: np.ndarray, epochs=1000, batch_size=32,
                                 cmpt.theta += dtheta
                                 cmpt.phi += dphi
 
+                    elif isinstance(layer, Activation) and isinstance(layer.nonlinearity, TrainableNonLinearity):
```

Comment:
please add a test case to `test_nonlinearities.py` to check that this gradient computation of the trainable nonlinearity works correctly